birdhouse/scripts/get-components-json.include.sh

```diff
@@ -0,0 +1,68 @@
+#!/bin/sh
+# Obtain a JSON representation of components enabled on this platform.
+#
+# Expected result should be similar to:
+# {
+#  "components": [
+#    "bird-house/birdhouse-deploy:components/monitoring",
+#    "bird-house/birdhouse-deploy:optional-components/canarie-api-full-monitoring",
+#    "bird-house/birdhouse-deploy:optional-components/all-public-access",
+#    "bird-house/birdhouse-deploy:optional-components/wps-healthchecks",
+#    "bird-house/birdhouse-deploy:optional-components/secure-thredds",
+#    "bird-house/birdhouse-deploy:optional-components/testthredds",
+#    "bird-house/birdhouse-deploy:optional-components/test-weaver",
+#    "bird-house/birdhouse-deploy:components/weaver",
+#    "bird-house/birdhouse-deploy:components/cowbird"
+#  ]
+# }
+#
+
+# default value in case of error or missing definitions
+export BIRDHOUSE_DEPLOY_COMPONENTS_JSON='{"components": []}'
+if [ -z "${EXTRA_CONF_DIRS}" ]; then
+  echo "No components in EXTRA_CONF_DIRS. Components JSON list will be empty!"
+  exit 0
+fi
+
+# resolve path considering if sourced or executed, and whether from current dir, pavics-compose include or another dir
+BIRDHOUSE_DEPLOY_COMPONENTS_ROOT=$(dirname -- "$(realpath "$0")")
+if [ "$(echo "${BIRDHOUSE_DEPLOY_COMPONENTS_ROOT}" | grep -cE "/birdhouse/?\$" 2>/dev/null)" -eq 1 ]; then
+  BIRDHOUSE_DEPLOY_COMPONENTS_ROOT=.
+else
+  BIRDHOUSE_DEPLOY_COMPONENTS_ROOT="${BIRDHOUSE_DEPLOY_COMPONENTS_ROOT}/.."
+fi
+cd "${BIRDHOUSE_DEPLOY_COMPONENTS_ROOT}" || ( \
+  echo "Error when attempting to move to [${BIRDHOUSE_DEPLOY_COMPONENTS_ROOT}]" && \
+  exit 1 \
+)
+
+# note: no quotes in 'ls' on purpose to expand glob patterns
+BIRDHOUSE_DEPLOY_COMPONENTS_LIST_KNOWN="$( \
+  ls -d1 ./*components/*/ \
+  | sed -E "s|\./(.*)/|\1|" \
+)"
+if [ -z "${BIRDHOUSE_DEPLOY_COMPONENTS_LIST_KNOWN}" ]; then
+  echo "[WARNING]" \
+    "Could not resolve known birdhouse-deploy components." \
+    "Aborting to avoid potentially leaking sensible details." \
+    "Components will not be reported on the platform's JSON endpoint."
+  exit 0
```

Comment:
Please do not `exit` at all !

This is being sourced by `pavics-compose.sh` so an `exit` here will exit in `pavics-compose.sh` and we do not want this since we do not want to leave the platform down during a unattended autodeploy.